### PR TITLE
FINERACT-1207: Clean up fineract-provider/config/swagger following in…

### PR DESCRIPTION
**Pull request by vidakovic**
## Description

1. Removed mustache templates in fineract-provider/config/swagger/templates
2. Removed fineract-provider/config/swagger/config.json.template; that was apparently just used for the command line generator before. "fineract-input.yaml.template is still there, because it's used in combination with the extracted "fineract.yaml"
3. Client documentation is updated

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

### CODE REVIEW

File: .gitignore
Comment (vorburger) [inline .gitignore]: @vidakovic this doesn't look right, is this intentional? Why?

File: docs/developers/swagger/client.md
Comment (vorburger) [inline docs/developers/swagger/client.md]:
```suggestion
The client JAR can be found in `fineract-client/build/libs`.
```

File: docs/developers/swagger/client.md
Comment (vorburger) [inline docs/developers/swagger/client.md]:
```suggestion
Note: Build only `fineract-client` may or may not actually save you some build time. There are still project module dependencies that might trigger a complete build.
```

File: docs/developers/swagger/client.md
Comment (vorburger) [inline docs/developers/swagger/client.md]: Oh, that's why the ignore. But... do we want this? Should we "support" that? Why? Should we just remove this entire section, to simplify things? I understand it's TECHNICALLY possible. I'm not sure we would want to deal with questions from anyone. @Grandolf49 any thoughts? @ptuomola any views?

File: fineract-client/build.gradle
Comment (vorburger) [inline fineract-client/build.gradle]: as above, let's see what we agree. Sorry I'm so adamant about not wanting to keep this simple.

File: fineract-provider/build.gradle
Comment (vorburger) [inline fineract-provider/build.gradle:604]: I didn't follow this in full details, but @ptuomola and/or @thesmallstar should review and sign off particularly on this change.. does that conversioning you wanted in the Swagger UI still work after this is removed?

**Review by vorburger — CHANGES_REQUESTED** [review]
I have reservations about "supporting" templates, and personally would just entirely remove that even more drastically. I'm happy to be convinced otherwise, if anyone wants to "defend" why we should do that.

File: .gitignore
Comment (vidakovic) [reply .gitignore]: @vorburger It is. That way you can still give advanced users the option to add their own templates if they need something custom we don't support. There's also a disclaimer in the client doc that we don't manage any custom templates.

I've added that reading the previous doc where this was described as some kind of feature. If we don't want that then I can remove again.

File: docs/developers/swagger/client.md
Comment (vidakovic) [reply docs/developers/swagger/client.md]: Done

File: docs/developers/swagger/client.md
Comment (vidakovic) [reply docs/developers/swagger/client.md]: Done

File: docs/developers/swagger/client.md
Comment (vidakovic) [reply docs/developers/swagger/client.md]: You tell me... I'm neither here nor their. Personally I tend more to explicitly providing generated clients like "normal" Java client, RxJava client, TypeScript client etc. (see SDK discussion). Otherwise people will start asking about their custom templates...

File: fineract-client/build.gradle
Comment (vidakovic) [reply fineract-client/build.gradle]: Yup, see my comments

File: fineract-provider/build.gradle
Comment (vidakovic) [reply fineract-provider/build.gradle:604]: I didn't see that this was copied to swagger-ui; I did see that there is a similar file there (though not sure why it's needed there). I could only see that config.json was used as parameter input for the command line generator.

File: docs/developers/swagger/client.md
Comment (vorburger) [reply docs/developers/swagger/client.md]: > You tell me... I'm neither here nor their. Personally I tend more to explicitly providing generated clients like "normal" Java client, RxJava client, TypeScript client etc. (see SDK discussion). Otherwise people will start asking about their custom templates...

Well, that makes 2 of us! Let's decide to remove the customizable templates. Our whole modularity project here is about people NOT forking and making local changes to Fineract (but be able to extend it), so why enable it here? I'd be happy to review and probably merge this PR once you have entirely removed the templates support.

File: fineract-client/build.gradle
Comment (vidakovic) [reply fineract-client/build.gradle]: FYI: removed all template related references (also from client.md).

**Review by vorburger — APPROVED** [review]
This LGTM. If @ptuomola can sign off on the remaining open point inline, we can merge this.

But Build Failure due to FINERACT-1167, for which we need to solve FINERACT-1168 - duh.

Let's Close and Reopen to rebuild.

**Comment by vorburger** [comment]
@ptuomola is this one OK for you, do you want to merge it? See open inline review point re rm that version thing...

File: fineract-provider/build.gradle
Comment (ptuomola) [reply fineract-provider/build.gradle:604]: Yes - this should be OK - the Swagger UI should only depend on the index.html that has been checked in, not on this config.json file. As far as I recall this is only used for the code generator...

**Review by ptuomola — APPROVED** [review]
LGTM

**Pull request merged**
